client/src/components/NotificationPopover.jsx

```diff
@@ -0,0 +1,83 @@
+import { Grid, Popover, makeStyles, Box, Button } from "@material-ui/core";
+import React, { useContext } from "react";
+import { userContext } from "../providers/UsersProvider";
+import BaseNotification from "./BaseNotification";
+const useStyles = makeStyles((theme) => ({
+  topBar: {
+    display: "flex",
+    alignItems: "center",
+    flexWrap: "wrap",
+    justifyContent: "space-between",
+  },
+}));
+
+const NotificationPopover = ({
+  handleCloseNotification,
+  anchorNotification,
+  notificationOpen,
+}) => {
+  const classes = useStyles();
+
+  const { notifications, localReadAllNotification, token } = useContext(
+    userContext
+  );
+
+  const useForceUpdate = () => {
+    const [, forceUpdate] = React.useState();
+
+    return React.useCallback(() => {
+      forceUpdate((s) => !s);
+    }, []);
+  };
+
+  const forceRender = useForceUpdate();
```

Comment:
Do you need to use this?
When a context changes, [it will force a re-render of all the components that consume that context](https://reactjs.org/docs/hooks-reference.html#usecontext).
If you absolutely need to use a `forceUpdate` function. It's better to implement it using `useReducer`. Here's that particular implementation from the [React docs](https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate):

```javascript
const [ignored, forceUpdate] = useReducer(x => x + 1, 0);

function handleClick() {
  forceUpdate();
}
```

But try to avoid doing this unless you absolutely need to.